File: fib.c
#define _POSIX_C_SOURCE 200809L
#include <stdio.h>
#include <stdlib.h>
#include <time.h>

double getTime(){
  struct timespec spec;
  clock_gettime(CLOCK_PROCESS_CPUTIME_ID, &spec);	
  double s = spec.tv_sec;
  double ms = spec.tv_nsec;
  return (s*1000 + ms / 1000000);
}

int fibRec(qianVal,zuoVal,rem){
  if(rem == 0){
    return zuoVal;
  }else{
    return fibRec(zuoVal,qianVal+zuoVal,rem-1);
  }
}

int fib(x){
  if(x < 2){
    return x;
  }else{
    return fibRec(0,1,x-1);
  }
}

int doWork(acc,rem){
  if (rem == 0) return acc;
  return doWork(acc + fib((acc+1) % 50), rem-1);
}
    
int main(int argc, char** argv){
  double startTime = getTime();
  printf("%d\n", doWork(0,atoi(argv[1])));
  double duration = getTime() - startTime;
  printf("LANGUAGE  C  %d\n", (int)duration);
  return 0;
}

File: fib.rs
extern crate time;

use std::os;
use time::precise_time_ns;

fn fib_rec(qianVal:i32,zuoVal:i32,rem:i32)->i32{
  if rem == 0{
    zuoVal
  }else{
    fib_rec(zuoVal,qianVal+zuoVal,rem-1)
  }
}

fn fib(x:i32)->i32{
  if x < 2{
    x
  }else{
    fib_rec(0,1,x-1)
  }
}

fn do_work(acc:i32,rem:i32)->i32{
  if rem == 0 {
    acc
  }else{
    do_work(acc + fib((acc+1) % 50), rem-1)
  }
}

fn main() {
  let n : i32 = from_str(os::args().get(1).as_slice()).expect("First argument must be an i32");
  let startTime = precise_time_ns();
  println!("{}",do_work(0,n));
  let duration = (precise_time_ns() - startTime) / 1000000;
  println!("LANGUAGE Rust {}", duration);
}
